### PR TITLE
[Testing] Add PhanTypeMismatchForeach rule to static analysis

### PR DESCRIPTION
**Pull request by johnsaigle**
This rule makes sure that when you pass a variable to foreach that the variable is expected to be something iterable. We had a few cases where parameters that were declared to be strings were actually arrays.

### CODE REVIEW

File: htdocs/feedback_mri_popup.php [hunk 1]
```diff
@@ -47,6 +47,9 @@
     $comments->setPredefinedComments($_POST['savecomments']['predefined']);
 
     // save all textual comments but only if there is an entry [sebas]
+    // The foreach below is operating on an array even though phan doesn't
```
Comment (zaliqarosli) [inline htdocs/feedback_mri_popup.php]: this seems hacky? why would phan think it was not an array if it was and if the rule is supposed to check exactly that?

Comment (zaliqarosli) [reply htdocs/feedback_mri_popup.php]: would defining an empty array variable, and then setting it to `$_POST['savecomments']['text']` bypass this?

Comment (johnsaigle) [reply htdocs/feedback_mri_popup.php]: I'll give it a shot!

Comment (johnsaigle) [reply htdocs/feedback_mri_popup.php]: No 😞

Comment (zaliqarosli) [reply htdocs/feedback_mri_popup.php]: :(

Comment (driusan) [reply htdocs/feedback_mri_popup.php]: We should always be using the most up-to-date version of phan (or just about any other library) that doesn't cause breakages in Travis..

With regard to this, I don't like the suppression either. Why can't you use `\Utility::asArray` to ensure foreach is working with an array?

Comment (johnsaigle) [reply htdocs/feedback_mri_popup.php]: I'll give it a try

Comment (johnsaigle) [reply htdocs/feedback_mri_popup.php]: Passes phan! Not really sure how to test it though

Comment (driusan) [reply htdocs/feedback_mri_popup.php]: Click on image level feedback in the imaging browser and make sure things still work/save in the popup.

Comment (johnsaigle) [reply htdocs/feedback_mri_popup.php]: Found some type errors while testing. Fixed in #4285. Will block this until that's merged.

**Comment by johnsaigle** [comment]
@zaliqarosli @driusan Everything seems to work so this is ready for re-review!

File: .phan/config.php [hunk 2]
```diff
@@ -30,7 +30,7 @@
 		"PhanUndeclaredClassMethod",
 		"PhanUndeclaredTypeParameter",
 		"PhanUndeclaredConstant",
-		"PhanTypeMismatchForeach",
+		"PhanTypeMismatchDefault",
```
Comment (driusan) [inline .phan/config.php]: This looks like it's just trading one ignored type for another. Why is MismatchDefault re-added?

Comment (johnsaigle) [reply .phan/config.php]: Oops bad rebase. I'll re-delete it.

File: htdocs/feedback_mri_popup.php (hunk 1 above)
Comment (zaliqarosli) [reply htdocs/feedback_mri_popup.php]: the comment on these lines can be removed now no?

**Pull request merged**